client/actions/index.js

```diff
@@ -60,26 +60,31 @@ export function setLoading (loading) {
 // }
 
 /**
- * A thunk that fetches all the events from the database and adds them to the store
+ * fetches Event[] then add events to store
+ * @returns {<Event[]>}
  */
 export function loadAllEvents () {
   return (dispatch) => {
-    // dispatch(setLoading(true))
-    return eventsData.showAllEvents().then((events) => {
-      // dispatch(setLoading(false))
-      // events.map((event) => {
-      //   event.disabled = false
-      // })
-      dispatch(setEvents(events))
-      return events
-    })
-    // .catch(err => {
-    //    dispatch(setLoading(false))
-    //    throw err
-    // })
+    dispatch(setLoading(true))
+    return eventsData
+      .showAllEvents()
+      .then((events) => {
+        dispatch(setLoading(false))
+        dispatch(setEvents(events))
+        return events
+      })
+      .catch((err) => {
+        dispatch(setLoading(false))
+        throw Error(err.message)
```

Comment:
```suggestion
        throw err
```

`throw new Error(err.message)` generates a new stacktrace, making it harder to find the source of the original issue.